sbol_utilities/helper_functions.py

```diff
@@ -105,6 +105,46 @@ def url_to_identity(url: str) -> str:
     return f'{split[0]}/{sbol3.string_to_display_id(split[1])}'
 
 
+def is_dna_part(obj: sbol3.Document) -> bool:
+    """Check if an SBOL Component is a DNA Part, i.e., having type 'SBO:DNA', and exactly 1 sequence property
+
+    :param obj: Sbol3 Document to be checked
+    :return: true if dna part
+    """
+    # must have only 1 type - of dna
+    def has_dna_type(x: sbol3.Document) -> bool:
+        if len(x.type) == 1:
+            return x.type.is_a(tyto.SBO.DNA)
+        return False
+
+    # all other roles, if present, must be terms of Sequence Ontology
```

Comment:
It can have non-SO terms, it just should have at least one SO term.

Reply:
Ohkay.